drivers/pinmux/stm32/pinmux_stm32g041x.h

```diff
@@ -0,0 +1,395 @@
+/*
+ * Copyright (c) 2020 Framework Computer - Kieran Levin <ktl@frame.work>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_DRIVERS_PINMUX_STM32_PINMUX_STM32G041_H_
+#define ZEPHYR_DRIVERS_PINMUX_STM32_PINMUX_STM32G041_H_
+
+/**
+ * alternate function mapping for STM32G041x8/xB
+ * @file Header for STM32G0 pin multiplexing helper
+ *
+ *
+ *
+ * **/
+
+#define STM32G0_PINMUX_FUNC_PA0_SPI2_SCK \
+	(STM32_PINMUX_ALT_FUNC_0 | STM32_OTYPER_PUSH_PULL | STM32_PUPDR_NO_PULL)
+#define STM32G0_PINMUX_FUNC_PA0_USART2_CTS \
+	(STM32_PINMUX_ALT_FUNC_1 | STM32_OTYPER_PUSH_PULL | STM32_PUPDR_NO_PULL)
+#define STM32G0_PINMUX_FUNC_PA0_TIM2_CH1_ETR \
+	(STM32_PINMUX_ALT_FUNC_2 | STM32_OTYPER_PUSH_PULL | STM32_PUPDR_NO_PULL)
+#define STM32G0_PINMUX_FUNC_PA0_LPTIM1_OUT \
+	(STM32_PINMUX_ALT_FUNC_5 | STM32_OTYPER_PUSH_PULL | STM32_PUPDR_NO_PULL)
+#define STM32G0_PINMUX_FUNC_PA1_SPI1_SCK__ \
+	(STM32_PINMUX_ALT_FUNC_0 | STM32_OTYPER_PUSH_PULL | STM32_PUPDR_NO_PULL)
+#define STM32G0_PINMUX_FUNC_PA1_I2S1_CK	\
+	(STM32_PINMUX_ALT_FUNC_1 | STM32_OTYPER_PUSH_PULL | STM32_PUPDR_NO_PULL)
+#define STM32G0_PINMUX_FUNC_PA1_USART2_RTS \
+	(STM32_PINMUX_ALT_FUNC_2 | STM32_OTYPER_PUSH_PULL | STM32_PUPDR_NO_PULL)
+#define STM32G0_PINMUX_FUNC_PA1__DE_CK \
+	(STM32_PINMUX_ALT_FUNC_3 | STM32_OTYPER_PUSH_PULL | STM32_PUPDR_NO_PULL)
+#define STM32G0_PINMUX_FUNC_PA1_TIM2_CH2 \
+	(STM32_PINMUX_ALT_FUNC_4 | STM32_OTYPER_PUSH_PULL | STM32_PUPDR_NO_PULL)
+#define STM32G0_PINMUX_FUNC_PA1_I2C1_SMBA \
+	(STM32_PINMUX_ALT_FUNC_8 | STM32_OTYPER_OPEN_DRAIN | STM32_PUPDR_NO_PULL)
+#define STM32G0_PINMUX_FUNC_PA1_EVENTOUT \
+	(STM32_PINMUX_ALT_FUNC_9 | STM32_OTYPER_PUSH_PULL | STM32_PUPDR_NO_PULL)
+#define STM32G0_PINMUX_FUNC_PA2_SPI1_MOSI__ \
+	(STM32_PINMUX_ALT_FUNC_0 | STM32_OTYPER_PUSH_PULL | STM32_PUPDR_NO_PULL)
+#define STM32G0_PINMUX_FUNC_PA2_I2S1_SD	\
+	(STM32_PINMUX_ALT_FUNC_1 | STM32_OTYPER_PUSH_PULL | STM32_PUPDR_NO_PULL)
+#define STM32G0_PINMUX_FUNC_PA2_USART2_TX \
+	(STM32_PINMUX_ALT_FUNC_2 | STM32_OTYPER_PUSH_PULL | STM32_PUPDR_NO_PULL)
+#define STM32G0_PINMUX_FUNC_PA2_TIM2_CH3 \
+	(STM32_PINMUX_ALT_FUNC_3 | STM32_OTYPER_PUSH_PULL | STM32_PUPDR_NO_PULL)
+#define STM32G0_PINMUX_FUNC_PA2_LPUART1_TX \
+	(STM32_PINMUX_ALT_FUNC_7 | STM32_OTYPER_PUSH_PULL | STM32_PUPDR_NO_PULL)
+#define STM32G0_PINMUX_FUNC_PA3_SPI2_MISO \
+	(STM32_PINMUX_ALT_FUNC_0 | STM32_OTYPER_PUSH_PULL | STM32_PUPDR_NO_PULL)
+#define STM32G0_PINMUX_FUNC_PA3_USART2_RX \
+	(STM32_PINMUX_ALT_FUNC_1 | STM32_OTYPER_PUSH_PULL | STM32_PUPDR_NO_PULL)
+#define STM32G0_PINMUX_FUNC_PA3_TIM2_CH4 \
+	(STM32_PINMUX_ALT_FUNC_2 | STM32_OTYPER_PUSH_PULL | STM32_PUPDR_NO_PULL)
+#define STM32G0_PINMUX_FUNC_PA3_LPUART1_RX \
+	(STM32_PINMUX_ALT_FUNC_6 | STM32_OTYPER_PUSH_PULL | STM32_PUPDR_NO_PULL)
+#define STM32G0_PINMUX_FUNC_PA3_EVENTOUT \
```

Comment:
To what peripheral EVENTOUT is referering to ?
Please remove pins for peripherals that are not supported yet on zephyr on STM32 globally (ok if not yet supported on G0)

Reply:
Not sure, but it is in the datasheet: 
![image](https://user-images.githubusercontent.com/734011/77411838-463c0e00-6d7a-11ea-9d67-335522df1ee5.png)
I would like to support all peripherals, as users may use the pinmux driver to setup gpio, but then reference LL code directly from their application. Why not support all alt mode functions? 
Also building a matrix of all supported alt mode functions and cross referencing them with supported peripherals is painful

Reply:
> I would like to support all peripherals, as users may use the pinmux driver to setup gpio, but then reference LL code directly from their application.

I got this point which is indeed valid. Though, we're anyway moving to a dts based pin definition, so this code should be here for too long (I hope). This being said this is an input
we can keep in mind for dts based definition.

> Why not support all alt mode functions?

Until now, we chose to restrain code to what could be tested in-tree (aside from the hal files that are tested externally). Reason is we don't want to perform reviews on back and forth changes, on things that could not be tested and maintained in tree.

Reply:
I removed all the event out definitions, but kept all the other alt definitions defined in the datasheet. Hopefully this can be merged in until the device tree configuration comes available. I do get your point that it is hard to test, and datasheets can have errors.